src/ghci/mod.rs

```diff
@@ -330,16 +325,29 @@ impl Ghci {
     /// This may fully restart the `ghci` process.
     #[instrument(skip_all, level = "debug")]
     pub async fn reload(&mut self, events: Vec<FileEvent>) -> miette::Result<()> {
+        for command in &self.opts.hooks.before_reload_ghci {
```

Comment:
should this be further down (inside the `if !actions.needs_reload.is_empty() {` expression)?